source/include/ble/cs_CrownstoneCentral.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Apr 15, 2021
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#include <ble/cs_UUID.h>
+#include <events/cs_EventListener.h>
+
+/**
+ * Class to connect to another crownstone, and write control commands.
+ *
+ * This class uses:
+ * - The EncryptionBuffer to read notifications to.
+ * - The CharacteristicReadBuffer to decrypt read data to.
+ * - The CharacteristicWriteBuffer to construct control packets.
+ */
+class CrownstoneCentral: EventListener {
+public:
+	cs_ret_code_t init();
+
+	/**
+	 * Connect, discover, and read session data.
+	 *
+	 * @param[in] stoneId              The stone ID of the crownstone you want to connect to.
+	 * @param[in] timeoutMs            How long (in milliseconds) to try connecting, before giving up.
+	 *
+	 * @return ERR_WRONG_STATE         When already connected.
+	 * @return ERR_BUSY                An operation is in progress (connect, write, disconnect).
+	 * @return ERR_WAIT_FOR_SUCCESS    When the connection will be attempted. Wait for EVT_CS_CENTRAL_CONNECT_RESULT.
+	 */
+	cs_ret_code_t connect(stone_id_t stoneId, uint16_t timeoutMs = 3000);
+
+	/**
+	 * Connect, discover, and read session data.
+	 *
+	 * @param[in] address              The MAC address of the crownstone you want to connect to.
+	 * @param[in] timeoutMs            How long (in milliseconds) to try connecting, before giving up.
+	 *
+	 * @return ERR_WRONG_STATE         When already connected.
+	 * @return ERR_BUSY                An operation is in progress (connect, write, disconnect).
+	 * @return ERR_WAIT_FOR_SUCCESS    When the connection will be attempted. Wait for EVT_CS_CENTRAL_CONNECT_RESULT.
+	 */
+	cs_ret_code_t connect(const device_address_t& address, uint16_t timeoutMs = 3000);
+
+	/**
+	 * Terminate current connection.
+	 *
+	 * @return ERR_SUCCESS             When already disconnected.
+	 * @return ERR_WAIT_FOR_SUCCESS    When disconnecting. Wait for EVT_BLE_CENTRAL_DISCONNECTED.
+	 */
+	cs_ret_code_t disconnect();
+
+	/**
+	 * Write a control command, and get the result.
+	 *
+	 * When the result code in the result data is ERR_WAIT_FOR_SUCCESS, you will get another EVT_CS_CENTRAL_WRITE_RESULT event.
+	 *
+	 * TODO: add timeout.
+	 *
+	 * @param[in] commandType          What control command.
+	 * @param[in] data                 Pointer to the command payload data.
+	 * @param[in] size                 Size of the command payload data.
+	 *
+	 * @return ERR_WAIT_FOR_SUCCESS    When the write is started. Wait for EVT_CS_CENTRAL_WRITE_RESULT.
+	 */
+	cs_ret_code_t write(cs_control_cmd_t commandType, uint8_t* data, uint16_t size);
+
+	/**
+	 * Request the write buffer, to avoid having to copy data to the write buffer.
```

Comment:
What does the user have to do with the write buffer?

Reply:
Added explanation.